1-Draft/RFCNNNN-Multi-Line-Continuation.md

```diff
@@ -0,0 +1,301 @@
+
+```PowerShell
+Get-ChildItem @-
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+-@
+```
+
+Backticks offer the advantage of continuing what they represent in PowerShell
+already (line continuation), and they are syntactically very similar to here-
+strings when used with the `@` symbol. That last point could be seen as a
+disadvantage as well, because they may be visually harder to distinguish from
+a single-quoted here-string.
+
+The `@-|-@` alternative doesn't pick up on the backtick for line continuation,
+but it is visually unique and easy to see in a script. The `-` character could
+be seen as representing the line that makes up the statement as well.
+
+### Inline splatting
+
+There has also been some discussion about the idea of inline splatting, using a
+format like `-@{...}` or `-@(...)`. Inline splatting has also been discussed
+separately on [RFC0002: Generalized Splatting](https://github.com/PowerShell/PowerShell-RFC/blob/master/2-Draft-Accepted/RFC0002-Generalized-Splatting.md), but using the syntax `@@{...}` or
+`@@(...)`.
+
+Here is an example showing what that might look like:
+
+```PowerShell
+Get-ChildItem -@{
+    LiteralPath = $rootFolder
+    File = $true
+    Filter = '*.ps*1'
+}
+```
+
+Using inline splatting to be able to span a single command across multiple
+lines like this has several limitations, including:
+
+1. You cannot transition to/from the inline splatted syntax without a bunch of
+manual tweaks to the command (either converting parameter syntax into hashtable
+or array syntax or vice versa).
+1. You're forced to choose between named parameters or positional
+parameters/arguments for each splatted collection. i.e. You can splat in a
+hashtable of named parameter/value pairs or an array of positional values, but
+you can't mix the two (the example shown just above is also used earlier in
+this RFC with positional parameters and switch parameters used without values,
+matching the way it is often used as a single-line command).
+1. There's no way to include unparsed arguments after the stop-parsing sigil in
+splatting. You can add it afterwards, but not include it within.
+1. Splatting requires a different syntax than typical parameter/argument input,
+which is more to learn. In contrast, the proposal above only requires learning
+about the `@` sigil (borrowed from splatting, but without specifying hashtables
+or arrays -- just allow all content until a newline), reducing the learning
+curve and allowing users to use parameters the same way in either case.
+1. Inline splatting attempts to resolve the issue for commands with arguments,
+but it does nothing for other scenarios where you want specific line wrapping
+other than the defaults that PowerShell implicitly supports.
+
+Further, unlike using a leading sigil such as `@`, which would work with
+Intellisense and tab expansion as they are coded now, inline splatting would
+require special work to make Intellisense and tab expansion work with it. That
```

Comment:
This is true, but we believe it's the right work to do: splatting should be better.

Reply:
Better, yes. Able to support splatting in members or the results of methods, absolutely, there's a PR for that. But when you push inline splatting as a way to make line wrapping better, you're violating the single-responsibility principle. Splatting is for command invocation with dynamic parameters that are chosen based on runtime execution. Also, line continuation is about more than just cmdlet or advanced function invocation. Take chained method invocation, for example, as shown earlier in this RFC. Also, I just remembered what I was trying to say about the stop-parsing sigil...with inline splatting as the way we wrap things, that's only good for cmdlet/advanced function parameters. If I want to wrap arguments passed after a stop-parsing sigil, I can't, no matter how long the line gets. But with this proposal, I would be able to do that.